addons/mail/models/mail_thread.py

```diff
@@ -1745,7 +1745,11 @@ def _message_post_process_attachments(self, attachments, attachment_ids, message
                     continue
                 if isinstance(content, str):
                     encoding = info and info.get('encoding')
-                    content = content.encode(encoding or 'utf-8')
+                    try:
```

Comment:
Mmmh, whatever you are trying to do, you should first begin with a test case ^^ (see test_mail_gateway.py notably for examples).